src/ray/raylet/node_manager.cc

```diff
@@ -519,6 +519,22 @@ ray::Status NodeManager::RegisterGcs() {
         [this] { local_object_manager_.FlushFreeObjects(); },
         RayConfig::instance().free_objects_period_milliseconds(),
         "NodeManager.deadline_timer.flush_free_objects");
+    periodical_runner_.RunFnPeriodically(
+        [this] {
+          // Trigger object spilling if current usage is above the specified threshold.
+          const float allocated_percentage =
+              static_cast<float>(local_object_manager_.GetPrimaryBytes()) /
+              object_manager_.GetMemoryCapacity();
+          if (allocated_percentage >= RayConfig::instance().object_spilling_threshold()) {
+            RAY_LOG(INFO) << "Triggering object spilling because current primary objects usage "
+                          << allocated_percentage * 100 << "% is above threshold "
+                          << RayConfig::instance().object_spilling_threshold() * 100
+                          << "%.";
+            local_object_manager_.SpillObjectUptoMaxThroughput();
+          }
+        },
+        RayConfig::instance().free_objects_period_milliseconds(),
```

Comment:
The previous code seems to have a return value

```
return GetLocalObjectManager().IsSpillingInProgress();
```

And now the store.cc cannot catch this state. Would there be any behavior change because of this? Looking at code, it seems like it shouldn't matter (seems like spill callback that's used by `spilling_required` doesn't use the return value), but I'd like to double check.

Reply:
I don't think there is any behavior change.